hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/Gateway.java

```diff
@@ -70,6 +75,17 @@ public Void call() throws Exception {
     OzoneConfigurationHolder.setConfiguration(ozoneConfiguration);
     UserGroupInformation.setConfiguration(ozoneConfiguration);
     loginS3GUser(ozoneConfiguration);
+
+    if (StringUtils.isEmpty(ozoneConfiguration.get(
+            OzoneConfigKeys.OZONE_METADATA_DIRS))) {
+      //Setting ozone.metadata.dirs if not set so that server setup doesn't
+      // fail.
+      Path tmpMetaDir = Files.createTempDirectory("ozone_s3g_tmpdir");
```

Comment:
Can this be set relative to the current working directory instead of `/tmp` which it would default to on most systems?

Reply:
done